### PR TITLE
Avoid nonzero sync in selected tensor paths

### PR DESCRIPTION
**Pull request by imxj**
## Summary
- avoid `nonzero()` in Janus image-token index discovery by collecting indices from the tokenizer output before tensorization
- replace Phi4 `adaptive_enc_mask` chunk lookup with `torch.bucketize`, avoiding the dense interval mask + `nonzero()` path
- replace chunked logprob GPU bool-mask + `torch.nonzero()` selection with a `torch.searchsorted` helper over sorted logprob indices

Refs #9889.

## Tests
- `PYTHONPATH=python .venv/bin/python -m pytest -q test/registered/unit/configs/test_janus_pro.py test/registered/unit/models/test_phi4mm_utils.py test/registered/unit/layers/test_logits_processor.py` -> 15 passed
- `PYTHONPATH=python .venv/bin/python -m pytest -q test/registered/unit/sampling/test_custom_logit_processor.py test/registered/unit/sampling/test_sampling_batch_info.py` -> 73 passed
- `.venv/bin/python scripts/ci/check_registered_tests.py`
- `git diff --check`

## B200 microbenchmark
On NVIDIA B200 with Torch 2.11.0+cu130, comparing the old chunked logprob bool-mask + `nonzero()` path to the new `searchsorted` helper:

- old: 0.0735 ms/iter
- new: 0.0459 ms/iter
- speedup: 1.60x

---
### CI States

Latest PR Test (Base): :x: **Missing `run-ci` label** — add it to run CI tests.
Latest PR Test (Extra): :x: **Blocked** — `run-ci` is required first.

### CODE REVIEW

**Comment by imxj** [comment]
Rebased this branch onto latest `main` (`89e501c`) to clear the out-of-date warning.

Local checks after rebase:
- `PYTHONPATH=python .venv/bin/python -m pytest -q test/registered/unit/configs/test_janus_pro.py test/registered/unit/models/test_phi4mm_utils.py test/registered/unit/layers/test_logits_processor.py` -> 15 passed
- `PYTHONPATH=python .venv/bin/python -m pytest -q test/registered/unit/sampling/test_custom_logit_processor.py test/registered/unit/sampling/test_sampling_batch_info.py` -> 73 passed
- `.venv/bin/python scripts/ci/check_registered_tests.py`
- `git diff --check`

The remaining PR Test gate failures are because the PR still has no `run-ci` label. I do not have permission to add that label directly.
